language/bytecode-verifier/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs

```diff
@@ -18,6 +18,13 @@ fn empty_module_no_errors() {
     basic_test_module().freeze().unwrap();
 }
 
+#[test]
+fn invalid_self_module_handle_index() {
+    let mut m = basic_test_module();
+    m.self_module_handle_idx = ModuleHandleIndex(12);
```

Comment:
I think we can have few more tests:
- No module handles exist, self_module_handle_idx = 0
- self_module_handle_idx = #module_handles - 1
- self_module_handle_idx = #module_handles
- self_module_handle_idx = some big number (not sure if we actually need this...)

Reply:
- No module handles exist, self_module_handle_idx = 0 (--> added in the fixup! commit)
- self_module_handle_idx = 0..#module_handles - 1 (--> handled in the prop tests)
- self_module_handle_idx = #module_handles | some big number (--> tested in the above test case)